src/C_TO_FSM.py

```diff
@@ -25,32 +25,14 @@ def C_AST_NODE_TO_C_CODE(c_ast_node, indent="", generator=None, is_lhs=False):
 
     # What nodes dont need semicolon?
     maybe_semicolon = ";"
-    if type(c_ast_node) == c_ast.Compound:
-        maybe_semicolon = ""
-    elif type(c_ast_node) == c_ast.If:
-        maybe_semicolon = ""
-    elif type(c_ast_node) == c_ast.While:
-        maybe_semicolon = ""
-    elif type(c_ast_node) == c_ast.For:
-        maybe_semicolon = ""
-    elif is_lhs:
+    if (
+        type(c_ast_node) == (c_ast.Compound or c_ast.If or c_ast.While or c_ast.For)
```

Comment:
Wow I didnt even know that syntax would work?

Reply:
@Datavenia how does this work? is c_ast types defined with or operators or something?

Reply:
You're correct that I got confused, closing this PR and I'll make a new one with proper syntax.